examples/performance-subscribe/src/main.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2022-2024 Contributors to the Eclipse Foundation
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import asyncio
+import json
+import logging
+import signal
+from datetime import datetime
+
+from sdv_model import Vehicle, vehicle  # type: ignore
+
+from velocitas_sdk.model import DataPoint
+from velocitas_sdk.util.log import (  # type: ignore
+    get_opentelemetry_log_factory,
+    get_opentelemetry_log_format,
+)
+from velocitas_sdk.vdb.reply import DataPointReply
+from velocitas_sdk.vehicle_app import VehicleApp
+
+# Configure the VehicleApp logger with the necessary log config and level.
+logging.setLogRecordFactory(get_opentelemetry_log_factory())
+logging.basicConfig(format=get_opentelemetry_log_format())
+logging.getLogger().setLevel("DEBUG")
+logger = logging.getLogger(__name__)
+
+GET_SPEED_REQUEST_TOPIC = "sampleapp/getSpeed"
+GET_SPEED_RESPONSE_TOPIC = "sampleapp/getSpeed/response"
+DATABROKER_SUBSCRIPTION_TOPIC = "sampleapp/currentSpeed"
+
+
+class PerformanceTestApp(VehicleApp):
+    """
+    A sample vehicle app that subscribes to a list of signals (from a file) and prints
+    the values + timestamps of the signals.
+    """
+
+    def __init__(self, vehicle_client: Vehicle):
+        super().__init__()
+        self.Vehicle = vehicle_client
+
+    async def on_start(self):
+        # The file path may need to be updated based on the location of the file.
+        test_signals_json = self.read_json("subscription_signals.json")
+        signals_json = test_signals_json["signals"]
+
+        for signal_json in signals_json:
+            signal_str = signal_json["path"]
+            await self.subscribe(signal_str)
+
+    def read_json(self, file_path: str):
+        with open(file_path, "r") as file:
+            return json.load(file)
+
+    async def subscribe(self, signal_str: str):
+        try:
+            node: DataPoint = self.Vehicle.getNode(signal_str)  # type: ignore
```

Comment:
Isn't there a type annotation on `Vehicle.getNode`? if so, the type annotation on `node` is obsolete

Reply:
There is, but sadly it is just "node" and not DataPoint but we know it always is a DataPoint (forced cast) and we need the "subscribe" method. IMO the Node method should also have a subscribe method and if it has children it should just subscribe to all children DataPoints.

```
def getNode(self, datapoint_str: str) -> Node:
        if self.get_path() not in datapoint_str:
            raise ValueError("Input string has to start with the root")
        nodes = datapoint_str.split(".")
        dataPoint = self

        if len(nodes) > 1:
            for node in nodes[1:]:
                try:
                    dataPoint = getattr(dataPoint, node)
                except Exception as err:
                    raise AttributeError("Node not found") from err

        return dataPoint
```

Reply:
Please create a follow up story for the subscribe on Node, I had this idea a while back as well.

Reply:
#127